tmt/steps/__init__.py

```diff
@@ -1123,42 +1131,55 @@ def login(context: Any, **kwargs: Any) -> None:
 
     @classmethod
     def plugins(cls, step: Step) -> List['Login']:
+        from tmt.steps.execute import ExecutePlugin
         """ Return list of login instances for given step """
         if not Login._enabled:
             return []
+        if cls._opt('test') and step.name == 'execute':
+            for p in step.phases(classes=(ExecutePlugin,)):
+                p.login_after_test = True
```

Comment:
I have had a similar idea since the beginning, but I didn't know where to place it. I moved to the constructor of `ExecutePlugin`. Is it the right location?

Reply:
Yes, this looks much better.